### PR TITLE
Add batch norm from scratch and with gluon

### PR DESCRIPTION
**Pull request by astonzhang**

### CODE REVIEW

**Comment by astonzhang** [comment]
Preview at:
https://github.com/astonzhang/gluon-tutorials-zh/blob/bn/chapter04_convolutional-neural-networks/batch-norm-scratch.md
https://github.com/astonzhang/gluon-tutorials-zh/blob/bn/chapter04_convolutional-neural-networks/batch-norm-gluon.md
